### PR TITLE
feat: update data-node methods to v2

### LINKED ISSUE #222: Update trading-data methods to v2
### Description

Currently bulk of methods in `data.py` and `data_raw.py` rely use [data-node/api/v1/trading-data.proto](https://docs.vega.xyz/testnet/api/grpc/data-node/api/v1/trading_data.proto) rather than [data-node/api/v2/trading-data.proto](https://docs.vega.xyz/testnet/api/grpc/data-node/api/v2/trading_data.proto).

As v1 will be deprecated soon, all methods using v1 need to be updated / replaced with v2 equivalent.

### PR DESCRIPTION
**Pull request by cdummett**
### Description
Updates market-sim functions to use data-node v2 GRPC requests.

`data.py`
- [x] positions_by_market 
- [x] party_account
- [x] find_market_id
- [x] find_asset_id
- [x] market_price_decimals
- [x] market_position_decimals
- [x] asset_decimals
- [x] best_prices
- [x] list_orders
- [ ] order_status_by_reference (REMOVED)
- [x] market_account
- [x] has_liquidity_provision
- [x] order_book_by_market

`raw_data.py`
- [x] positions_by_market
- [x] all_markets
- [x] market_info
- [x] asset_info
- [x] all_market_accounts
- [x] market_accounts
- [x] market_data
- [x] infrastructure_fee_accounts
- [x] order_status
- [x] order_status_by_reference
- [x] market_depth
- [x] liquidity_provisions

`trading.py`
- [x] submit_order

`governance.py`
- [x] _proposal_loader
- [x] _get_blockchain_time
- [x] propose_market_from_config
- [x] propose_future_market
- [x] propose_network_parameter_change
- [x] propose_market_update
- [x] propose_asset
- [x] _build_generic_proposal 
- [x] _make_and_wait_for_proposal

`trading.py`
- [x] submit_order

`service.py`
- [x] market_price_decimals
- [x] market_pos_decimals
- [x] asset_decimals
- [x] market_to_asset
- [x] create_asset
- [x] create_market_from_config
- [x] create_simple_market
- [x] submit_order
- [x] update_network_parameter
- [x] settle_market
- [x] party_account
- [x] positions_by_market
- [x] all_markets
- [x] market_info
- [x] market_data
- [x] infrastructure_fee_accounts
- [x] market_accounts
- [x] best_prices
- [x] order_book_by_market
- [x] market_depth
- [x] submit_simple_liquidity
- [x] has_liquidity_provision
- [x] submit_liquidity
- [x] find_market_id
- [x] find_asset_id
- [x] order_status
- [x] order_status_by_reference (REMOVED)
- [x] order_status_from_feed
- [x] start_order_monitoring
- [x] list_orders
- [x] order_book_by_market
- [x] get_blockchain_time
- [x] margin_levels
- [x] get_trades

### Testing
Passing all tests locally.

No mentions of `trading_data_client` (v1) in `service.py`, `VegaTradingDataClient` (v1)  in api files, or `data_node_protos` (v1) in any files

### Breaking Changes
None.

### Closes
Closes #222

### CODE REVIEW

File: vega_sim/service.py [hunk 1]
```diff
@@ -1292,30 +1292,7 @@ def order_status(
             Optional[vega.Order], the requested Order object or None if nothing found
         """
         return data_raw.order_status(
-            order_id=order_id, data_client=self.trading_data_client, version=version
-        )
-
-    def order_status_by_reference(
```
Comment (TomMcL) [inline vega_sim/service.py:1298]: Can we not keep this but just use the new query pattern?

**Review by TomMcL — COMMENTED** [review]
Query about the _by_reference function but looks good other than that. At some point we should go through and kill all the `_v2` references but that's safer as a second pass.

File: vega_sim/service.py (hunk 1 above)
Comment (cdummett) [reply vega_sim/service.py:1298]: I removed this one as it wasn't used anywhere in the codebase and the v1 method `OrderByReference` doesn't have a direct equivalent in v2.

The closest v2 method is `ListOrders` which already has a method in `service.py` called `list_orders`.

 I have just realised the `service.list_orders` methods is missing an optional `reference` arg to pass through to `data.list_orders` method though. If you are happy to leave out `order_status_by_reference` and just use `list_orders` I'll add that arg in?

Comment (TomMcL) [reply vega_sim/service.py:1298]: Yeah that sounds good to me 👍

File: vega_sim/service.py [hunk 2]
```diff
@@ -1445,6 +1445,7 @@ def list_orders(
         wallet_name: str,
         key_name: str,
         market_id: str,
+        reference: Optional[str] = None,
         live_only: Optional[bool] = True,
     ) -> List[data.Order]:
         """Return a list of orders for the specified market and party.
```
Comment (TomMcL) [inline vega_sim/service.py:1451]: docstring

**Review by TomMcL — APPROVED** [review]
lgtm

**Pull request merged**